### PR TITLE
View product by id

### PR DESCRIPTION
**Pull request by mohamed1refaie**

### CODE REVIEW

File: src/main/java/com/piper/valley/controllers/ProductController.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    ///////////////////////////////////////*  CONTROLLER ACTION  *///////////////////////////////////////////
+
+    //@PreAuthorize("hasAuthority('ADMIN')")
+    @RequestMapping(value = "/product/view/{id}", method = RequestMethod.GET)
+    public ModelAndView viewProduct(@PathVariable("id") long id) {
+        Optional<Product> product = productService.getProductById(id);
+        if (!product.isPresent()) {
+            return new ModelAndView("error/404");
+        }
+        return new ModelAndView("product/view", "product", product);
```
Comment (wewark) [inline src/main/java/com/piper/valley/controllers/ProductController.java]: Send product.get() instead, to avoid the ".get()" everywhere in the HTML file.

Comment (SherifAbdlNaby) [reply src/main/java/com/piper/valley/controllers/ProductController.java]: point

**Comment by SherifAbdlNaby** [comment]
Khaly adding Product ye3mlk redirect 3ala view el product dah, fe tare2a hatkhlek t-return el latest added ID, w beh te-redirect 3ala el correct ID.

**Comment by mohamed1refaie** [comment]
Done

**Pull request merged**